### ISSUE TITLE
Recover from embedding errors

### ISSUE THREAD

**Issue by mario-guerra** [Azure-Samples/azure-search-openai-demo#1394]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

I encountered an error with the document embedding step due to something that was malformed in my input documents. The error came from the OpenAI embedding endpoint, and because the exception was not handled it stopped the entire data ingestion process. 

I added some error handling logic to get past this issue, and I'm providing a patch with the changes I made since others may encounter this issue as well. I tried submitting a PR, but I don't have access to commit to this repo.

```
diff --git a/scripts/prepdocslib/embeddings.py b/scripts/prepdocslib/embeddings.py
index d6193d1..c2fc9c6 100644
--- a/scripts/prepdocslib/embeddings.py
+++ b/scripts/prepdocslib/embeddings.py
@@ -7,7 +7,7 @@ import tiktoken
 from azure.core.credentials import AzureKeyCredential
 from azure.core.credentials_async import AsyncTokenCredential
 from azure.identity.aio import get_bearer_token_provider
-from openai import AsyncAzureOpenAI, AsyncOpenAI, RateLimitError
+from openai import AsyncAzureOpenAI, AsyncOpenAI, RateLimitError, BadRequestError
 from tenacity import (
     AsyncRetrying,
     retry_if_exception_type,
@@ -94,8 +94,13 @@ class OpenAIEmbeddings(ABC):
                 before_sleep=self.before_retry_sleep,
             ):
                 with attempt:
-                    emb_response = await client.embeddings.create(model=self.open_ai_model_name, input=batch.texts)
-                    embeddings.extend([data.embedding for data in emb_response.data])
+                    try:
+                        emb_response = await client.embeddings.create(model=self.open_ai_model_name, input=batch.texts)
+                        embeddings.extend([data.embedding for data in emb_response.data])
+                    except BadRequestError as e:
+                        print(f"Error creating embeddings for batch: {e}. Moving on to the next batch.")
+                        # Add None to embeddings list for each text in the current batch
+                        embeddings.extend([None] * len(batch.texts))
                     if self.verbose:
                         print(f"Batch Completed. Batch size  {len(batch.texts)} Token count {batch.token_length}")
 
diff --git a/scripts/prepdocslib/searchmanager.py b/scripts/prepdocslib/searchmanager.py
index 4f9e14b..68417f0 100644
--- a/scripts/prepdocslib/searchmanager.py
+++ b/scripts/prepdocslib/searchmanager.py
@@ -216,7 +216,11 @@ class SearchManager:
                         texts=[section.split_page.text for section in batch]
                     )
                     for i, document in enumerate(documents):
-                        document["embedding"] = embeddings[i]
+                        if embeddings[i] is not None:
+                            document["embedding"] = embeddings[i]
+                        else:
+                            print(f"Warning: No embedding for document {i}. Assigning default value.")
+                            document["embedding"] = [0.1] * 1536
                 if image_embeddings:
                     for i, (document, section) in enumerate(zip(documents, batch)):
                         document["imageEmbedding"] = image_embeddings[section.split_page.page_num]

```

**Comment by pamelafox** [comment]
I chatted with Mario offline. It looks like Mario got the "$.input is invalid" error described here:
https://stackoverflow.com/questions/77542684/openai-error-invalidrequesterror-while-trying-to-create-an-embedding

My guess is that our splitting strategy is probably failing on some data situation, and creating chunks that our too large.

I think we have to replicate this issue so that we can debug the root cause, so that developers don't have to set a default vector embedding (which will negatively impact vector search).

**Comment by singloudly90** [comment]
@pamelafox I encountered this issue too , but i changed to text-embedding-3-large then it would be running successfully.
You are right that the chunks might be too large, because even it passed thru the embedding,

 it would be failed at Azure Search 

TypeError: azure.search.documents._generated.aio.operations._documents_operations.DocumentsOperations.index() got multiple values for keyword argument 'error_map'_

that only happens in my large documents which around 14mb.

**Comment by pamelafox** [comment]
This is now fixed! Please pull main or changes from #1423
